### ISSUE TITLE
Popover hides when hovering over popover itself

### ISSUE THREAD

**Issue by rodrigoreis22** [mgcrea/angular-strap#959]
I want to have some behavior on a popover triggered on hover. For example, when user hover on a user, a popover with basic info and a Follow button will appear.
Is there a way to accomplish this?

**Comment by afc163** [comment]
```js
/**
 * Popover Directive.  Wraps the AngularStrap popover and
 *     1) Allows the popover to stay open when hovering / focusing the contents.
 *        If the popover was only triggered by 'hover' or 'focus', it closes as soon
 *        as the mouse leaves or focus is lost on the triggering element.
 *
 *     2) Lazy initializes the popover until it's needed.  This reduces the
 *        amount of scope.
 *
 *     3) When the popover closes, cleans up the AngularStrap popover memory leak
 *        caused by using directives for popover content.
 */
.directive('accessibleContentsPopover', ['$popover', '$timeout', function ($popover, $timeout) {
    return {
        restrict: 'A',
        //scope: true,
        link: function (scope, elem, attrs) {
            var options = {},
                popover,
                popoverElement,
                openPopoverTimeout,
                closePopoverTimeout;
 
            // Move the attributes into our own
            // object and link the scopes together
            angular.extend(options, attrs);
            options.scope = scope;
 
            function cancelTimers() {
                if (openPopoverTimeout) {
                    $timeout.cancel(openPopoverTimeout);
                }
 
                if (closePopoverTimeout) {
                     $timeout.cancel(closePopoverTimeout);
                }
            }
 
            function unbindPopoverElementEvents() {
                if (popoverElement) {
                    popoverElement.off('mouseenter').off('focusin').off('mouseleave').off('focusout');
                }
            }
 
            function destroyPopover() {
                if (popover) {
                    popover.hide(); // allow animation leave
                    popover = undefined;
                }
                popoverElement = undefined;
            }
 
            function delayClose() {
                // Cancel any other timers
                cancelTimers();
 
                // Start the clock
                closePopoverTimeout = $timeout(function () {
                    unbindPopoverElementEvents();
                    destroyPopover();
                }, 250);
            }
 
            function openPopover(delay) {
                cancelTimers();
 
                // Create the popover
                if (!popover) {
                    openPopoverTimeout = $timeout(function () {
                        popover = $popover(elem, options);
 
                        // Could be using an external template, need to wrap in promise
                        popover.$promise.then(function () {
                            // Show the popover after digest
                            scope.$$postDigest(function() {
                              popover.show();
 
                              // Grab a reference to the popover contents
                              popoverElement = popover.$element;
 
                              // When mousing or focusing inside the contents,
                              // cancel the closing of the popover
                              popoverElement.on('mouseenter', cancelTimers);
                              popoverElement.on('focusin', cancelTimers);
 
                              // When leaving the popover's contents, start the
                              // close process
                              popoverElement.on('mouseleave', delayClose);
                              popoverElement.on('focusout', delayClose);
                          });
                        });
                    }, delay);
                }
            }
 
            // On mouseenter or focus, open the popover
            elem.on('mouseenter', function () { openPopover(250); });
            elem.on('focus', function () { openPopover(750); });
 
            // When we leave the trigger element
            // start the close process
            elem.on('mouseleave', delayClose);
            elem.on('blur', delayClose);
 
            // Clean up any timers and event handlers
            // when the directive is destroyed
            scope.$on('$destroy', function () {
                cancelTimers();
                elem.off('mouseenter').off('focus').off('mouseleave').off('blur');
                unbindPopoverElementEvents();
                destroyPopover();
            });
        }
    };
}])
```

**Comment by joaquimds** [comment]
I used this solution but had a problem where the popover would still respond to the data-trigger attribute, as well as the mousenter / focus / mouseleave / blur events in this solution. In order to fix this, I added elem.off('click') immediately after popover.show in the openPopover(delay) function.

**Comment by iFytil** [comment]
@joaquimds 

I've experienced this issue. Instead of `elem.off('click')`, I simply set the trigger to manual to avoid unwanted behavior from click/mouse events.

`var options = {trigger: 'manual'}`

The reason I have a different solution is because I have some elements which have both a popover on-hover (using the above directive) as well as an unrelated ng-click behavior on the element triggering the popover. Your solution would prevent the ng-click from working properly.
